WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -55,4 +65,190 @@ class WordPressAuthenticatorTests: XCTestCase {
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginFromPresenter(presenterSpy, animated: true)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testShowLoginForJustWPComPresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testSignInForWPOrgReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPOrg()
+
+        XCTAssertTrue(vc is LoginSiteAddressViewController)
+    }
+
+    func testShowLoginForJustWPComSetsMetaProperties() throws {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy,
+                                                     xmlrpc: "https://example.com/xmlrpc.php",
+                                                     username: "username",
+                                                     connectedEmail: "email-address@example.com")
+
+        let navController = try XCTUnwrap(presenterSpy.presentedVC as? LoginNavigationController)
+        let controller = try XCTUnwrap(navController.viewControllers.first as? LoginEmailViewController)
+
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertEqual(controller.loginFields.restrictToWPCom, true)
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogXMLRPC, "https://example.com/xmlrpc.php")
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogUsername, "username")
+        XCTAssertEqual(controller.loginFields.username, "email-address@example.com")
+    }
+
+    func testShowLoginForSelfHostedSitePresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForSelfHostedSite(presenterSpy)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testSignInForWPComReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPCom()
+
+        XCTAssertTrue((vc as Any) is LoginEmailViewController)
```

Comment:
Would the same form used in the other tests work here, too?

```suggestion
        XCTAssertTrue(vc is LoginEmailViewController)
```

Reply:
That one is a little weird.  Where the other vc is Type tests are testing functions that return a super class of the view controller being checked, with the function signinForWPCom() it returns a LoginEmailViewController or fatal errors.  So to make sure this test does anything I had to convert the VC to any or it would throw up a warning that the is check always succeeds.  

Since there isn't much going on in this test, I think removing it is probably the best option.  As you mentioned with the source tags in a different comment, there isn't really any logic being tested here.  I went ahead and removed it, but what do you think?  Should it be there?

Reply:
Thanks for the extra context. In light of that, then yes **removing the test was the best thing to do** 👍 

If the type system is already ensuring for us that the return value is `LoginEmailViewController`, then there is no added value in writing a test for it, because the compiler will already enforce that.